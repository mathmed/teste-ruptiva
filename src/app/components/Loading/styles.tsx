import styled from 'styled-components/native';
import { FontAwesome5 as Icon } from '@expo/vector-icons';

export const Container = styled.View`
  margin-top: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const LoadingText = styled.Text`
  color: #ff9000;
  font-weight: bold;
  margin-top: 16px;
  font-size: 16px;
`;

export const LoadingIcon = styled(Icon)`
  transform: rotate(45deg);
`;
