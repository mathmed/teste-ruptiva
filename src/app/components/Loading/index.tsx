import React from 'react';
import { Container, LoadingText, LoadingIcon } from './styles';
import { Animated, Easing } from 'react-native';

interface IconCustomProps {
  text: string;
}

const Loading: React.FC<IconCustomProps> = ({ text, ...props }) => {
  const spinValue = new Animated.Value(0);
  Animated.loop(
    Animated.timing(spinValue, {
      toValue: 1,
      duration: 2000,
      easing: Easing.linear,
      useNativeDriver: false,
    }),
  ).start();

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });

  return (
    <Container>
      <Animated.View style={{ transform: [{ rotate: spin }] }}>
        <LoadingIcon
          style={{ transform: [{ rotateY: '180deg' }] }}
          size={36}
          color="#ff9000"
          name="spinner"
          {...props}
        />
      </Animated.View>
      <LoadingText>{text}</LoadingText>
    </Container>
  );
};

export default Loading;
