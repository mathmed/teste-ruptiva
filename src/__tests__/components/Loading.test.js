import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../../app/components/Loading';

let screen = null;
jest.useFakeTimers();

describe('[COMPONENTS] Loading', () => {
  beforeEach(async () => {
    screen = await renderer.create(<Loading text={'Carregando'} />);
  });
  it('Should render corretly', async () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
