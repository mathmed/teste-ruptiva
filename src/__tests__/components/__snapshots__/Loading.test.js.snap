// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] Loading Should render corretly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "marginTop": 50,
      },
    ]
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#ff9000",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 16,
        },
      ]
    }
  >
    Carregando
  </Text>
</View>
`;
